### PR TITLE
Add additional examples of usage

### PR DESCRIPTION
**Pull request by davkean**
Add additional examples of usage, and fixed the generic method usage which looked incorrect in markdown.

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md [hunk 1]
```diff
@@ -36,15 +36,33 @@ namespace N
         public void BannedMethod(int i) {}
 
         public void BannedMethod<T>(T t) {}
+
+        public void BannedMethod<T>(Func<T> f) {}
```
Comment (mavasani) [inline src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md]: Thanks. While you are touching this file, might be good to add that you can get the documentation ID format for any declaration by compiling sample code with that declaration with XML documentation comments [enabled](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/doc-compiler-option) and then opening the generated xml file.

Comment (mavasani) [reply src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md]: Seems that each declaration needs to be a public API with an xml doc comment to have its documentation ID generated. For example, compiling the below code with `csc /t:library /doc:ClassLibrary.xml ClassLibrary.cs`:

```csharp
using System;

namespace N
{
    /// <summary>
    /// 
    /// </summary>
    public class BannedType
    {
        /// <summary>
        /// 
        /// </summary>
        public BannedType() { }

        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public int BannedMethod() { return 0; }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="i"></param>
        public void BannedMethod(int i) { }

        /// <summary>
        /// 
        /// </summary>
        /// <typeparam name="T"></typeparam>
        /// <param name="t"></param>
        public void BannedMethod<T>(T t) { }

        /// <summary>
        /// 
        /// </summary>
        /// <typeparam name="T"></typeparam>
        /// <param name="f"></param>
        public void BannedMethod<T>(Func<T> f) { }

        /// <summary>
        /// 
        /// </summary>
        public string BannedField;

        /// <summary>
        /// 
        /// </summary>
        public string BannedProperty { get; }

        /// <summary>
        /// 
        /// </summary>
        public event EventHandler BannedEvent;
    }

    /// <summary>
    /// 
    /// </summary>
    /// <typeparam name="T"></typeparam>
    public class BannedType<T>
    {
    }
}
```

generates the following xml doc file:

```
<?xml version="1.0"?>
<doc>
    <assembly>
        <name>ClassLibrary</name>
    </assembly>
    <members>
        <member name="T:N.BannedType">
            <summary>
            
            </summary>
        </member>
        <member name="M:N.BannedType.#ctor">
            <summary>
            
            </summary>
        </member>
        <member name="M:N.BannedType.BannedMethod">
            <summary>
            
            </summary>
            <returns></returns>
        </member>
        <member name="M:N.BannedType.BannedMethod(System.Int32)">
            <summary>
            
            </summary>
            <param name="i"></param>
        </member>
        <member name="M:N.BannedType.BannedMethod``1(``0)">
            <summary>
            
            </summary>
            <typeparam name="T"></typeparam>
            <param name="t"></param>
        </member>
        <member name="M:N.BannedType.BannedMethod``1(System.Func{``0})">
            <summary>
            
            </summary>
            <typeparam name="T"></typeparam>
            <param name="f"></param>
        </member>
        <member name="F:N.BannedType.BannedField">
            <summary>
            
            </summary>
        </member>
        <member name="P:N.BannedType.BannedProperty">
            <summary>
            
            </summary>
        </member>
        <member name="E:N.BannedType.BannedEvent">
            <summary>
            
            </summary>
        </member>
        <member name="T:N.BannedType`1">
            <summary>
            
            </summary>
            <typeparam name="T"></typeparam>
        </member>
    </members>
</doc>
```

Comment (davkean) [reply src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md]: What's with the double-backtick on generic methods? I've tested this without it and it works, and it appears taht `()` is optional on methods without parameters. Looks like something is parsing them?

Comment (mavasani) [reply src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md]: We use the public APIs exposed from Microsoft.CodeAnalysis:

https://github.com/dotnet/roslyn/blob/e704ca635bd6de70a0250e34c4567c7a28fa9f6d/src/Compilers/Core/Portable/DocumentationCommentId.cs#L84-L87

https://github.com/dotnet/roslyn/blob/e704ca635bd6de70a0250e34c4567c7a28fa9f6d/src/Compilers/Core/Portable/DocumentationCommentId.cs#L45-L49

Comment (sharwell) [reply src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md]: > What's with the double-backtick on generic methods?

Single backtick is for generic parameters of types. Double backtick is generic parameters of methods. Examples:

```csharp
class OuterClass<TOuter> {
  class InnerClass<TInner> {
    // M:OuterClass`1.InnerClass`1.Method``1(System.Func{`0})
    void Method<TMethod>(Func<TOuter> outer) { }
    // M:OuterClass`1.InnerClass`1.Method``1(System.Func{`1})
    void Method<TMethod>(Func<TInner> outer) { }
    // M:OuterClass`1.InnerClass`1.Method``1(System.Func{``0})
    void Method<TMethod>(Func<TMethod> method) { }
  }
}
```

Comment (mavasani) [reply src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md]: I'll merge the PR. We can enhance the documentation further in a follow-up PR.

**Pull request merged**